frame/cosmwasm/src/lib.rs

```diff
@@ -1050,6 +1074,50 @@ pub mod pallet {
 			Ok(())
 		}
 
+		/// Create an empty iterator.
+		pub(crate) fn do_db_scan(vm: &mut CosmwasmVM<T>) -> Result<u32, CosmwasmVMError<T>> {
+			let iterator_id =
+				CurrentIteratorId::<T>::increment().map_err(|_| Error::<T>::IteratorIdOverflow)?;
+			ContractStorageIterators::<T>::insert(
+				vm.contract_info.clone(),
+				iterator_id,
+				None::<BoundedVec<_, _>>,
+			);
+			Ok(iterator_id)
+		}
+
+		/// Return the next (reversed-key, value) pair and save the state. If the next key
+		/// is `None`, the iterator is removed from the storage.
+		pub(crate) fn do_db_next(
+			vm: &mut CosmwasmVM<T>,
+			iterator_id: u32,
+		) -> Result<Option<(Vec<u8>, Vec<u8>)>, CosmwasmVMError<T>> {
+			let storage_key_entry =
+				ContractStorageIterators::<T>::try_get(&vm.contract_info, &iterator_id)
+					.map_err(|_| Error::<T>::IteratorNotFound)?;
```

Comment:
Why don't we use the VM extension to store the iterator as the lifetime of the iterator cannot exceed the lifetime of the executing contract?